src/pydvl/influence/frameworks/twice_differentiable.py

```diff
@@ -0,0 +1,54 @@
+from abc import ABC
+from typing import Callable, Generic, Optional, Sequence, Tuple, TypeVar
+
+from numpy.typing import NDArray
+
+TensorType = TypeVar("TensorType", bound=Sequence)
+ModelType = TypeVar("ModelType")
+
+
+class TwiceDifferentiable(ABC, Generic[TensorType, ModelType]):
+    def __init__(
+        self,
+        model: ModelType,
+        loss: Callable[[TensorType, TensorType], TensorType],
+    ):
+        pass
+
+    def num_params(self) -> int:
+        pass
+
+    def split_grad(
+        self,
+        x: TensorType,
+        y: TensorType,
+        progress: bool = False,
+    ) -> NDArray:
+        """
+        Calculate the gradient of the model wrt each input x and labels y.
+        The output is therefore of size [Nxp], with N the amout of points (the
+        length of x and y) and P the number of parameters.
+        """
+        pass
+
+    def grad(self, x: TensorType, y: TensorType) -> Tuple[NDArray, TensorType]:
+        """
+        It calculates the gradient of model parameters with respect to input x
```

Comment:
A pronoun is strange here. The subject is just beside the verb. ("grad(...): calculates"). There are more examples.

```suggestion
        Calculates the gradient of model parameters with respect to input x
```